### PR TITLE
DOC: Update `daal4py` .md instructions file

### PR DESCRIPTION
**Pull request by david-cortes-intel**
## Description

_Add a comprehensive description of proposed changes_

_List associated issue number(s) if exist(s): #6 (for example)_

_Documentation PR (if needed): #1340 (for example)_

_Benchmarks PR (if needed): https://github.com/IntelPython/scikit-learn_bench/pull/155 (for example)_

---

Daal4Py is now merged into the package scikit-learn-intelex, but its docs have not been updated to reflect this situation.

This PR updates the .md file under the daal4py root folder to:
* Make the relationship betweel d4p and sklearnex more clear.
* Remove outdated support table (now it's the same as sklearnex).
* Link to sklearnex instructions whenever possible.
* Correct the list of packages that are needed for GPU and MPI support.

Checklist to comply with **before moving PR from draft**:

**PR completeness and readability**

- [x] I have reviewed my changes thoroughly before submitting this pull request.
- [x] I have updated the documentation to reflect the changes or created a separate PR with update and provided its number in the description, if necessary.
- [x] Git commit message contains an appropriate signed-off-by string _(see [CONTRIBUTING.md](https://github.com/uxlfoundation/scikit-learn-intelex/blob/main/CONTRIBUTING.md#pull-requests) for details)_.
- [x] I have added a respective label(s) to PR if I have a permission for that.
- [x] I have resolved any merge conflicts that might occur with the base branch.

**Testing**

Not applicable.

**Performance**

Not applicable.

### CODE REVIEW

**Review by yuejiaointel — APPROVED** [review]
LGTM!

**Review by Vika-F — APPROVED** [review]
